src/DynamicRupture/Output/Builders/ReceiverBasedOutputBuilder.cpp

```diff
@@ -0,0 +1,183 @@
+    }
+    {
+      auto* memorySpace = outputData->stressFaceAlignedToGlb[receiverId].data();
+      RotationMatrixViewT rotationMatrixView(memorySpace, {6, 6});
+      symmetricTensor2RotationMatrix(faceNormal, tangent1, tangent2, rotationMatrixView, 0, 0);
+    }
+    {
+      auto faceAlignedToGlb =
+          init::T::view::create(outputData->faceAlignedToGlbData[receiverId].data());
+      auto glbToFaceAligned =
+          init::Tinv::view::create(outputData->glbToFaceAlignedData[receiverId].data());
+
+      seissol::model::getFaceRotationMatrix(
+          faceNormal, tangent1, tangent2, faceAlignedToGlb, glbToFaceAligned);
+    }
+  }
+}
+
+void ReceiverBasedOutputBuilder::initOutputVariables(
+    std::array<bool, std::tuple_size<DrVarsT>::value>& outputMask) {
+  auto assignMask = [&outputMask](auto& var, int receiverId) {
+    var.isActive = outputMask[receiverId];
+  };
+  misc::forEach(outputData->vars, assignMask);
+
+  auto allocateVariables = [this](auto& var, int) {
+    var.maxCacheLevel = outputData->maxCacheLevel;
+    var.allocateData(this->outputData->receiverPoints.size());
+  };
+  misc::forEach(outputData->vars, allocateVariables);
+}
+
+void ReceiverBasedOutputBuilder::initJacobian2dMatrices() {
+  const auto& faultInfo = meshReader->getFault();
+  const auto& verticesInfo = meshReader->getVertices();
+  const auto& elementsInfo = meshReader->getElements();
+
+  size_t nReceiverPoints = outputData->receiverPoints.size();
+  outputData->jacobianT2d.resize(nReceiverPoints);
+
+  for (size_t receiverId = 0; receiverId < nReceiverPoints; ++receiverId) {
+    auto side = outputData->receiverPoints[receiverId].localFaceSideId;
+    auto elementIndex = outputData->receiverPoints[receiverId].elementIndex;
+
+    assert(elementIndex >= 0);
+
+    const auto& element = elementsInfo[elementIndex];
+    auto face = getGlobalTriangle(side, element, verticesInfo);
+
+    VrtxCoords xab, xac;
+    {
+      constexpr size_t x{0}, y{1}, z{2};
+      xab[x] = face.p2[x] - face.p1[x];
+      xab[y] = face.p2[y] - face.p1[y];
+      xab[z] = face.p2[z] - face.p1[z];
+
+      xac[x] = face.p3[x] - face.p1[x];
+      xac[y] = face.p3[y] - face.p1[y];
+      xac[z] = face.p3[z] - face.p1[z];
+    }
+
+    auto faultIndex = outputData->receiverPoints[receiverId].faultFaceIndex;
+    auto* tangent1 = faultInfo[faultIndex].tangent1;
+    auto* tangent2 = faultInfo[faultIndex].tangent2;
+
+    Eigen::Matrix<real, 2, 2> matrix;
+    matrix(0, 0) = MeshTools::dot(tangent1, xab);
+    matrix(0, 1) = MeshTools::dot(tangent2, xab);
+    matrix(1, 0) = MeshTools::dot(tangent1, xac);
+    matrix(1, 1) = MeshTools::dot(tangent2, xac);
+    outputData->jacobianT2d[receiverId] = matrix.inverse();
+  }
+}
+
+void ReceiverBasedOutputBuilder::assignNearestInternalGaussianPoints() {
+  auto& geoPoints = outputData->receiverPoints;
+  constexpr int numPoly = CONVERGENCE_ORDER - 1;
+
+#ifdef dunavant
```

Comment:
We never define `dunavant`. Can you add this flag to `CMakeLists.txt`?

Reply:
Hi, yes I know. I was surprised that it was added before.  Ok. I will do it.

Reply:
Added the following to Cmake
https://github.com/SeisSol/SeisSol/blob/1356811a4e3dc612c337812c4fbd9408f975bdea/CMakeLists.txt#L325-L331